static/js/containers/App.js

```diff
@@ -1,17 +1,38 @@
 // @flow
 import React from "react"
-import { Route } from "react-router"
+import { Switch, Route } from "react-router"
+import urljoin from "url-join"
+
+import { routes } from "../lib/urls"
+
+import Header from "../components/Header"
+
+import HomePage from "./pages/HomePage"
+import LoginPage from "./pages/LoginPage"
+
 import type { Match } from "react-router"
 
-export default class App extends React.Component<*, void> {
-  props: {
-    match: Match
-  }
+type Props = {
+  match: Match
+}
 
+export default class App extends React.Component<Props, void> {
   render() {
+    const { match } = this.props
     return (
       <div className="app">
-        <Route path="/" render={() => <h1>MIT xPro</h1>} />
+        <Header />
+        <Switch>
+          <Route
+            exact
+            path={`${match.url}${routes.home}`}
+            component={HomePage}
+          />
```

Comment:
Something to note ahead of time: the course catalog PR sets the root URL to use the course catalog Django view instead of using the React-friendly `index` view. We'll probably want to do one of the following:
1. Set a new route for the course catalog (`/catalog` et al) and link to it in this new home page
1. Get rid of this home page and the header, add a login link on the current course catalog page, and just navigate to the root URL when login completes
1. Go with the hybrid approach – create a new bundle and add the React header component to the existing course catalog page which is rendered in a Django template

(1) seems to be the easiest out, but we'll probably want (3) some time soon.

Reply:
Yeah, I figured as much, I was basically punting on this decision since we don't have a good idea where things would live and for working on this branch I needed some kind of placeholder for "/"

Reply:
This home page route is still "/", so we will still see this React home page after the login flow, but when the browser is refreshed, the course catalog will be loaded again. I don't have any major problems with that since the UI has no specs at all yet, but worth pointing out